specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/JobAgents.json

```diff
@@ -290,13 +296,52 @@
           "$ref": "../../../common/v1/types.json#/definitions/Sku",
           "description": "The name and tier of the SKU."
         },
+        "identity": {
+          "$ref": "#/definitions/JobAgentIdentity",
+          "description": "The identity of the job agent."
+        },
         "properties": {
           "$ref": "#/definitions/JobAgentProperties",
           "description": "Resource properties.",
           "x-ms-client-flatten": true
         }
       }
     },
+    "JobAgentIdentity": {
```

Comment:
>JobAgentIdentity [](http://example.com/codeflow?start=5&length=16)

Please see if you can refactor this part to define it using definitions from common-types instead

e.g. with

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/managedidentity.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/managedidentity.json)

or

[https://github.com/Azure/azure-rest-api-specs/blob/5dd1107d5f2be8d600325d795450e1d854fbe7e8/specification/common-types/resource-management/v5/types.json#L201](https://github.com/Azure/azure-rest-api-specs/blob/5dd1107d5f2be8d600325d795450e1d854fbe7e8/specification/common-types/resource-management/v5/types.json#L201)

Reply:
We would like to keep this consistent with other Microsoft.Sql APIs such as: https://github.com/Azure/azure-rest-api-specs/blob/def187e2e78d7173d8fdd7f77740dd9719e1dfbf/specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/Databases.json#L875C7-L875C7

Reply:
Either define a common version of identity types under Microsoft.Sql RP or refer it from ARM Common Types. Don't define the same type for all different resource types separately in their swagger json files.

Reply:
Just in case you want to define your own RP common types, I strongly recommend having them in your own 'common.json' or similar name json file, to help everyone remember that they're *shared* contracts (not to be broken), and also avoid accidental circular dependencies between JSON files, which some tools can't handle today.

Reply:
aligning with SQL common type def needs extra efforts from SQL security team. We have to ask for an exception in current version. we will coordinate with SQL security team to fix it in next version. But for this version, please unblock.
Here is the work item to align common types: https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/2697929

Reply:
Is it possible to set an ETA and link it to a Feature/user Story to track this task? Just want to make sure that the next person who makes changes for next API version, is able to address this.

Reply:
The item has been added to SQL API committer task queue and will be fixed in the next API version.